src/static_handler.rs

```diff
@@ -168,25 +168,43 @@ impl Cache {
         };
 
         let if_modified_since = match req.headers.get::<IfModifiedSince>().cloned() {
-            None => return Ok(self.response_with_cache(path, last_modified_time)),
+            None => return Ok(self.response_with_cache(req, path, last_modified_time)),
             Some(IfModifiedSince(HttpDate(time))) => time.to_timespec(),
         };
 
         if last_modified_time <= if_modified_since {
             Ok(Response::with(status::NotModified))
         } else {
-            Ok(self.response_with_cache(path, last_modified_time))
+            Ok(self.response_with_cache(req, path, last_modified_time))
         }
     }
 
-    fn response_with_cache<P: AsRef<Path>>(&self, path: P, modified: Timespec) -> Response {
+    fn response_with_cache<P: AsRef<Path>>(&self, req: &mut Request, path: P, modified: Timespec) -> Response {
         use iron::headers::{CacheControl, LastModified, CacheDirective, HttpDate};
+        use iron::headers::{ContentLength, ContentType};
+        use iron::method::Method;
+        use iron::mime::{Mime, TopLevel, SubLevel};
+        use iron::modifiers::Header;
 
-        let mut response = Response::with((status::Ok, path.as_ref()));
+        let mut response: Response;
         let seconds = self.duration.as_secs() as u32;
         let cache = vec![CacheDirective::Public, CacheDirective::MaxAge(seconds)];
+        let metadata = fs::metadata(path.as_ref()).unwrap();
```

Comment:
I am on vacation until Monday but will attend these issues when I can.
Thanks for the comments!
On Fri, Jul 29, 2016 at 11:42 Markus Unterwaditzer <notifications@github.com>
wrote:

> In src/static_handler.rs
> <https://github.com/iron/staticfile/pull/57#discussion_r72841677>:
>
> >          let seconds = self.duration.as_secs() as u32;
> >          let cache = vec![CacheDirective::Public, CacheDirective::MaxAge(seconds)];
> > +        let metadata = fs::metadata(path.as_ref()).unwrap();
>
> In case of lacking permissions I would return a 500 Internal Server Error.
> However, in such a case it would be useful to return an Err(IronError)
> instead of Ok(Response) which changes the type signature of this function.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/iron/staticfile/pull/57/files/9d8dc78e30dd21386a09e7740f724b45af7f4ce7#r72841677>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ACirdi0kMJC8SJ1TzZXsJFoDaF6Lmezuks5qakmsgaJpZM4GWPDZ>
> .
>